lib/Backend/amd64/LowererMDArch.cpp

```diff
@@ -1036,35 +1092,14 @@ LowererMDArch::GetArgSlotOpnd(uint16 index, StackSym * argSym, bool isHelper /*=
 
     IRType type = argSym ? argSym->GetType() : TyMachReg;
     const bool isFloatArg = IRType_IsFloat(type) || IRType_IsSimd128(type);
-    RegNum reg = RegNOREG;
-
-    if (!isFloatArg && argPosition <= IntArgRegsCount)
-    {
-        switch (argPosition)
-        {
-#define REG_INT_ARG(Index, Name)    \
-        case ((Index) + 1):         \
-            reg = Reg ## Name;      \
-            break;
-#include "RegList.h"
-        default:
-            Assume(UNREACHED);
-        }
-    }
-    else if (isFloatArg && argPosition <= XmmArgRegsCount)
+    RegNum reg = GetRegFromArgPosition(isFloatArg, argPosition);
+#ifndef _WIN32
+    if (reg != RegNOREG && this->xplatCallArgs.IsRecording())
```

Comment:
Seems all calls to `GetArgSlotOpnd` is to load an argument, and if we are to record every one, `IsRecording()` should be true.